offchain/README.md

```diff
@@ -1,8 +1,119 @@
+Refer to their own documentation for more details.
+
+#### Configuration
+
+It is possible to configure the behavior of any of the projects by passing CLI arguments and using environment variables.
+Execute the following command to check the available options for each project:
+
+```shell
+cargo run -- -h
+```
+
+##### Redis TLS Configuration
+
+To connect the Broker to a Redis server via TLS, the server's URL must use the `rediss://` scheme (with two "s"es).
 This is currently the only way to tell `Broker` to use a TLS connection.
+
+### Tests
+
+To run the tests available in any of the projects, execute the command:
+
+```shell
+cargo test
+```
+
+For integration tests, it is possible to see additional logs by setting the following variable:
+
+```shell
+export CARTESI_TEST_VERBOSE=1
+```
+
+## Architecture Overview
+
+The Cartesi Node is an event-based solution that may be depicted as follows:
+
+![Node architecture diagram](./docs/node-architecture.drawio.svg)
+
+The **Broker** is a Redis-based message broker that mediates the transferring of *Inputs*, *Outputs* and *Claims* between the Cartesi Node components.
+
+The [**State-fold Server**](./state-server/README.md) is the component responsible for tracking blockchain state changes and deriving *Inputs* from them to be processed.
+It's also responsible for submitting *Claims* to the blockchain whenever they are created by the Cartesi Node.
+
+The [**Dispatcher**](./dispatcher/README.md) is the component that messages *Inputs* via the **Broker** to be processed elsewhere and captures **Claims** to be submitted by the **State-fold Server**.
+
+The [**Advance Runner**](./advance-runner/README.md) is the one responsible for relaying *Inputs* to the **Server Manager** to be processed by the underlying DApp running in the embedded **Cartesi Machine** and for capturing wathever **Outputs** that may be generated as a result.
+
+The [**Indexer**](./indexer/README.md) consumes all *Inputs* and *Outputs* transferred by the **Broker** and store them in a **PostgreSQL** database for later querying via a **GraphQL Server** instance.
+
+The [**Inspect Server**](./inspect-server/README.md) is responsible for the processing of *Inspect* requests via HTTP.
+
+## Workflows
+
+The Cartesi Node may act as a *User* (aka *Reader*) or *Validator* Node, hence performing different roles, which may be represented by different workflows.
+
+### Reader Flow
+
+In the Reader Flow, any *Input* generated by the **State-fold Server** is received by the **Dispatcher** and relayed to the **Broker** via *rollups-inputs-stream*.
+The *Input* is eventually consumed by the **Advance Runner** and used to advance the state of the **Server Manager**, thus generating an *Advance* request to be processed by the underlying DApp.
+
+After finishing the processing, the DApp may genrerate a number of *Outputs*, which are eventually retrieved by the **Advance Runner** and fed back into the **Broker** via *rollups-outputs-stream*.
```

Comment:
```suggestion
After finishing the processing, the DApp may generate a number of *Outputs*, which are eventually retrieved by the **Advance Runner** and fed back into the **Broker** via *rollups-outputs-stream*.
```

Reply:
IMHO we should omit the names of the event streams for simplicity's sake or link to more detailed info in the `rollups-events` README, for example. It's odd to mention them by name and provide no further details.